src/qamqpchannel.cpp

```diff
@@ -238,6 +252,39 @@ void QAmqpChannelPrivate::qosOk(const QAmqpMethodFrame &frame)
     Q_EMIT q->qosDefined();
 }
 
+/*! Report and change state. */
+void QAmqpChannelPrivate::newState(ChannelState state)
+{
+    qAmqpDebug() << "Channel"
+                 << name
+                 << "state:"
+                 << channelState
+                 << "->"
+                 << state;
+    channelState = state;
+}
+
+QDebug operator<<(QDebug dbg, QAmqpChannelPrivate::ChannelState s)
+{
+    switch(s) {
+        case QAmqpChannelPrivate::ChannelClosedState:
```

Comment:
nitpick on the style guide, but the ```case``` statements should be aligned with ```switch```, so back 1 indentation level: http://wiki.qt.io/Qt_Coding_Style#Switch_statements